File: app/src/main/java/com/s16/engmyan/fragment/CreditFragment.java
package com.s16.engmyan.fragment;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.webkit.WebView;

import com.s16.app.ActionBarUtils;
import com.s16.engmyan.Constants;
import com.s16.engmyan.R;

public class CreditFragment extends DialogFragment {

	@SuppressLint("InlinedApi")
	@Override
	public void onCreate(Bundle savedInstanceState) {
	    super.onCreate(savedInstanceState);
	    setStyle(DialogFragment.STYLE_NO_TITLE, R.style.AppTheme_Translucent);
	}
	
	@Override
	public Dialog onCreateDialog(Bundle savedInstanceState) {
		Dialog dialog = super.onCreateDialog(savedInstanceState);
		dialog.getWindow().requestFeature(Window.FEATURE_NO_TITLE);
		return dialog;
	}
	
	@SuppressLint("SetJavaScriptEnabled")
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
			Bundle savedInstanceState) {
		
		ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.fragment_credit, container, false);
		
		WebView webView = (WebView)rootView.findViewById(R.id.webViewCredit);
		webView.getSettings().setAllowFileAccess(true);
		webView.getSettings().setJavaScriptEnabled(true);
		webView.loadUrl(Constants.URL_CREDIT);
		
		View actionClose = rootView.findViewById(R.id.closeButton);
		ActionBarUtils.getInstance(getContext()).makeActionButton(actionClose);
		actionClose.setOnClickListener(new View.OnClickListener() {
			
			@Override
			public void onClick(View v) {
				dismiss();
			}
		});
		
		return rootView;
	}
}

File: app/src/main/java/com/s16/engmyan/fragments/CreditFragment.kt
package com.s16.engmyan.fragments

import android.app.Activity
import android.graphics.Point
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import androidx.fragment.app.DialogFragment
import com.s16.engmyan.Constants
import com.s16.engmyan.R
import com.s16.widget.setPositiveButton
import kotlinx.android.synthetic.main.fragment_credit.*


class CreditFragment : DialogFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        dialog?.requestWindowFeature(Window.FEATURE_NO_TITLE);
        return inflater.inflate(R.layout.fragment_credit, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        webViewCredit.settings.allowFileAccess = true
        webViewCredit.loadUrl(Constants.URL_CREDIT)

        dialogButtons.setPositiveButton(android.R.string.ok) { _, _ ->
            dialog!!.cancel()
        }
    }

    override fun onResume() {
        super.onResume()

        val height = (getScreenHeight(requireActivity()) * 0.95).toInt()
        dialog?.window?.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, height)
    }

    private fun getScreenHeight(activity: Activity): Int {
        val size = Point()
        activity.windowManager.defaultDisplay.getSize(size)
        return size.y
    }

    companion object {
        @JvmStatic
        fun newInstance() = CreditFragment()
    }
}
